test/metabase/test/data/impl/get_or_create.clj

```diff
@@ -147,6 +148,60 @@
   (data-perms/set-database-permission! (perms-group/all-users) new-db-id :perms/create-queries :query-builder-and-native)
   (data-perms/set-database-permission! (perms-group/all-users) new-db-id :perms/download-results :one-million-rows))
 
+(defn- preprocess-dbdef-for-fks
+  "Return vector of form [[table-name field-name fk-table-name]...]"
+  [dbdef]
+  (loop [[table-def & table-defs] (:table-definitions dbdef)
+         result []]
+    (if (nil? table-def)
+      result
+      (recur table-defs
+             (into result
+                   (comp (filter (comp some? :fk))
+                         (map #(vector (:table-name table-def)
+                                       (:field-name %)
+                                       (name (:fk %)))))
+                   (:field-definitions table-def))))))
+
+(defn- dbdef->fk-field-infos
+  "Generate `fk-field-infos` structure. It is a seq of maps of 2 keys: :id and :fk-target-field-id. Existing database,
+  tables and fields in app db are examined to get the required info."
```

Comment:
Have you considered returning a map from ids to target-ids?

Reply:
That would make the code more succinct, but I'm leaning towards this version as key names follow what's in app db so it is evident from glance on:
```
{:id (get-in table-name->field-name->field [table-name field-name :id])
 :fk-target-field-id (get-in table-name->pk-field [target-table-name :id])}
```
what are the returned data. If that does not make sense lmk, I'll change that.